jardin/cache_stores/s3.py

```diff
@@ -19,18 +20,24 @@ def __init__(self, bucket_name, path="", delete_expired_files=False):
         self.bucket_name = bucket_name
         self.path = f"{path}/jardin_cache"
         self.delete_expired_files = delete_expired_files
+        self._s3 = None
+        self._lock = Lock() # guards lazy S3 init
 
-    @memoized_property
+    @property
     def s3(self):
-        # Instantiate the S3 resource lazily to provide greater flexibility when mocking with 'moto'.
+        # Instantiate the S3 client lazily to provide greater flexibility when mocking with 'moto'.
         # Typically you wouldn't need to do this, but Jardin's cache system instantiates this class
         # at import-time which can cause initialization problems for 'moto'. We side-step the problem
-        # by deferring the creation of the S3 resource to first usage instead of instance init.
-        return boto3.resource('s3')
+        # by deferring the creation of the S3 client to first usage instead of instance init.
+        if self._s3 is None:
+            with self._lock:
+                if self._s3 is None:
+                    self._s3 = boto3.client('s3')
+        return self._s3
 
     def check_valid_bucket(self):
```

Comment:
we do not call this method anywhere right ? You still want to keep for debugging ?

Reply:
Correct. I just pushed a commit to remove it.